pkg/openstack/credentials.go

```diff
@@ -41,33 +42,43 @@ type Credentials struct {
 }
 
 // GetCredentials computes for a given context and infrastructure the corresponding credentials object.
-func GetCredentials(ctx context.Context, c client.Client, secretRef corev1.SecretReference) (*Credentials, error) {
+func GetCredentials(ctx context.Context, c client.Client, secretRef corev1.SecretReference, allowDNSKeys bool) (*Credentials, error) {
 	secret, err := extensionscontroller.GetSecretByReference(ctx, c, &secretRef)
 	if err != nil {
 		return nil, err
 	}
-	return ExtractCredentials(secret)
+	return ExtractCredentials(secret, allowDNSKeys)
 }
 
 // ExtractCredentials generates a credentials object for a given provider secret.
-func ExtractCredentials(secret *corev1.Secret) (*Credentials, error) {
+func ExtractCredentials(secret *corev1.Secret, allowDNSKeys bool) (*Credentials, error) {
+
+	var altDomainNameKey, altTenantNameKey, altUserNameKey, altPasswordKey, altAuthURLKey *string
+	if allowDNSKeys {
+		altDomainNameKey = pointer.String(DNSDomainName)
+		altTenantNameKey = pointer.String(DNSTenantName)
+		altUserNameKey = pointer.String(DNSUserName)
+		altPasswordKey = pointer.String(DNSPassword)
+		altAuthURLKey = pointer.String(DNSAuthURL)
+	}
+
 	if secret.Data == nil {
 		return nil, fmt.Errorf("secret does not contain any data")
 	}
-	domainName, err := getRequired(secret, DomainName)
+	domainName, err := getRequired(secret, DomainName, altDomainNameKey)
 	if err != nil {
 		return nil, err
 	}
-	tenantName, err := getRequired(secret, TenantName)
+	tenantName, err := getRequired(secret, TenantName, altTenantNameKey)
 	if err != nil {
 		return nil, err
 	}
-	userName := getOptional(secret, UserName)
-	password := getOptional(secret, Password)
-	applicationCredentialID := getOptional(secret, ApplicationCredentialID)
-	applicationCredentialName := getOptional(secret, ApplicationCredentialName)
-	applicationCredentialSecret := getOptional(secret, ApplicationCredentialSecret)
-	authURL := getOptional(secret, AuthURL)
+	userName := getOptional(secret, UserName, altUserNameKey)
+	password := getOptional(secret, Password, altPasswordKey)
+	applicationCredentialID := getOptional(secret, ApplicationCredentialID, nil)
+	applicationCredentialName := getOptional(secret, ApplicationCredentialName, nil)
+	applicationCredentialSecret := getOptional(secret, ApplicationCredentialSecret, nil)
```

Comment:
Maybe we should allow the DNS keys also for these three new parameters?
See https://github.com/gardener/external-dns-management/blob/master/pkg/controller/provider/openstack/handler.go#L78-L80

Reply:
Thanks, I think this was introduced recently in `external-dns-management`, for some reason I had a slightly older version and didn't see them, so I thought they are not supported for DNS for some reason. Was going to ask you about it but it slipped my mind, good that you noticed it.

Reply:
Done (in a new commit)